java/common/common-task/src/main/java/sleeper/common/task/RunDataProcessingTasks.java

```diff
@@ -53,28 +57,63 @@
  * Finds the number of messages on a queue, and starts up one EC2 or Fargate task for each, up to a
  * configurable maximum.
  */
-public class RunCompactionTasks {
-    private static final Logger LOGGER = LoggerFactory.getLogger(RunCompactionTasks.class);
+public class RunDataProcessingTasks {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RunDataProcessingTasks.class);
 
     private final InstanceProperties instanceProperties;
     private final TaskCounts taskCounts;
     private final CompactionTaskHostScaler hostScaler;
     private final TaskLauncher taskLauncher;
+    private CompactionECSLaunchType launchType;
+    private String containerName;
+    private String clusterName;
+    private String fargateDefinitionFamily;
 
-    public RunCompactionTasks(
-            InstanceProperties instanceProperties, EcsClient ecsClient, AutoScalingClient asClient, Ec2Client ec2Client) {
-        this(instanceProperties,
-                () -> ECSTaskCount.getNumPendingAndRunningTasks(instanceProperties.get(COMPACTION_CLUSTER), ecsClient),
-                EC2Scaler.create(instanceProperties, asClient, ec2Client),
-                (numberOfTasks, checkAbort) -> launchTasks(ecsClient, instanceProperties, numberOfTasks, checkAbort));
+    public RunDataProcessingTasks(
+            InstanceProperties instanceProperties,
+            EcsClient ecsClient,
+            AutoScalingClient asClient,
+            Ec2Client ec2Client,
+            CompactionTaskType compactionTaskType) {
+
+        this.instanceProperties = instanceProperties;
+
+        // Initialize taskCounts lambda
+        this.taskCounts = () -> ECSTaskCount.getNumPendingAndRunningTasks(
+                instanceProperties.get(compactionTaskType == CompactionTaskType.BULK_EXPORT ? BULK_EXPORT_CLUSTER : COMPACTION_CLUSTER),
```

Comment:
This is checking the same condition twice, as it's also checked in the if statement below. This can lead to confusion because you expect to see all the behaviour that's dependent on it in one place.

You could resolve which instance property should be used before the definition of this lambda. If you have a private constructor with all the parameters, and a public static method for each task type, you can decide which instance property should be used in the public static method for the task type.